### DISCUSSION TITLE
Version 1.9

### DISCUSSION THREAD

**Discussion by ksqsf** [rimeinn/rime-moran#44] [Announcements]
**Full Changelog**: https://github.com/ksqsf/rime-moran/compare/1.2...1.9

2.0版本尚未完全就緒，暫緩發佈。 **請將 Ver 1.9 理解爲 2.0 的預覽版本，是大版本更新。**

## 碼表變更

+ **新增**：單字表（`moran.chars.dict.yaml`）現已擴充到 8 萬餘字，爲音形碼之最。
+ **移除**：刪除了 `aabc` 碼表和方案。
+ **變更**：更新詞語詞庫；修正錯音錯字錯碼。面向 librime 1.11 的長詞聯想功能做了少量詞庫優化。
+ **變更**：**簡體簡碼碼表大量優化。（如不習慣，建議保留自己原本碼表。）**

## 功能變更

+ **新增**：拆分濾鏡功能，使用 `Ctrl+i` 切換。
<img width="656" alt="image" src="https://github.com/ksqsf/rime-moran/assets/23358293/fc913855-c3aa-4757-8282-e2006a3cc287">

+ **新增**：全碼新增 `yyxx/` 打法。

<img width="97" alt="image" src="https://github.com/ksqsf/rime-moran/assets/23358293/3bcd6990-ebad-4d17-a9d1-bc871f1263b4">

+ **新增**：輔篩模式（`moran_aux`）。可在輸入末尾打一位或兩位輔助碼以篩選第一個詞語。請注意，該模式依賴於最新版本的 Lua 插件！如果上屏後無法清空輔助碼，說明 Lua 版本不足。使用例：

https://github.com/ksqsf/rime-moran/assets/23358293/c018a7ff-b51f-4f29-b1f8-39b304f865e3

+ **新增**：並擊輸入模式（`moran_bj`）。

+ **新增**：單字輔助碼提示功能，可顯示一個字的所有輔助碼打法。該功能默認關閉。請參考 `moran.schema.yaml` 文件尾 `moran/enable_aux_hint` 的說明。（性能較差，不建議使用。）
<img width="342" alt="image" src="https://github.com/ksqsf/rime-moran/assets/23358293/95bef381-5776-495f-9a3d-2e9410d94dbb">

+ **新增**：簡碼提示功能。可顯示字或詞的簡碼。該功能默認關閉。請參考 `moran.schema.yaml` 文件尾 `moran/enable_quick_code_hint` 的說明。
<img width="405" alt="image" src="https://github.com/ksqsf/rime-moran/assets/23358293/e14acd9e-9cc2-42fd-876f-87895cad18cc">

+ **修正**：修正了分號處理器可能重複觸發的問題。

+ **修正**：修正了一些性能問題和兼容性問題。

## 前端相關變更

+ **macOS**：鼠鬚管默認主題修改爲 `wechat` 和 `wechat_dark`。
+ **Android**：改善了 Android 同文輸入法兼容性，現在無需增添額外文件，直接導入配置即可使用。
+ **iOS**：現已登陸 iOS 「倉」輸入法的內置方案倉庫。打開倉輸入法設置界面，點擊「輸入方案設置」，點擊右上方「➕」號，再點擊「方案下載」，即可直接下載安裝「魔然〈繁〉」「魔然〈簡〉」。

目前，本方案經測試，可正常運作於目前所有 Rime 前端平臺。爲確保正常使用，請確保使用 librime v1.10 或以上版本，Lua 插件版本也應當與之匹配。

使用效果請移步[此 Wiki 頁面](https://github.com/ksqsf/rime-moran/wiki/Gallery)。

## 配置系統變更及相應的升級說明

⚠️ 本版本對原本的配置系統做了較大修改。因此，比較穩妥的升級方式是 (1) 備份好個人數據後 (2) 全新安裝一套配置 (3) 重新導入原來的自定義，如用戶詞庫、碼表、選項設置等。

重要的用戶數據文件列表：
+ `*.userdb` 文件夾：使用過程中產生的用戶詞庫 ← 強烈建議備份！
+ `moran_custom_phrases.txt`：用戶自定義短語文件
+ 其他任何手動修改過的詞庫文件，請檢查 `moran.extended.dict.yaml` 、 `moran_fixed.dict.yaml` 、 `moran_fixed_simp.dict.yaml` 。

本次更新中，將原本散落在各 schema 中的拼寫運算和快捷鍵設置合併到了 `moran.yaml` 文件中。要設置新的快捷鍵，或想要引入新的拼寫運算，請參考 moran.yaml 中的說明。具體用例可參見 #41 。

由於 Rime 配置系統較複雜，如有疑問，請發起 issue 或 discussion 探討。

## 致謝

本版本得到了（排名不分先後） Kelatte 、三山剑客、SunShine、吕小布、吥捨洳楓、 gaboolic、小幽幽等的大力協助。在此表示感謝！

<hr /><em>This discussion was created from the release <a href='https://github.com/ksqsf/rime-moran/releases/tag/1.9'>Version 1.9</a>.</em>

**Comment by bosbyj** [comment]
版本日志好严谨啊，谢谢您的付出🙏

**Comment by liangkaichun** [comment]
“iOS：現已登陸 iOS 「倉」輸入法的內置方案倉庫。打開倉輸入法設置界面，點擊「輸入方案設置」，點擊右上方「➕」號，再點擊「方案下載」，即可直接下載安裝「魔然〈繁〉」「魔然〈簡〉」。”

手机上使用了仓输入法，安装的是简体方案，那些不常用的生僻字都还显示成“？”，这个有办法过滤掉吗？我配置中选择了GBK，看起来没有起作用。

**Reply by ksqsf** [reply]
仓没有打包charcode插件, 因此charset filter无效 (https://github.com/imfuxiao/LibrimeKit/issues/3). 这个问题计划之后改用 lua 版本的 charset filter 解决.

**Reply by liangkaichun** [reply]
了解了，谢谢！

**Reply by ksqsf** [reply]
实现了 lua 版本的 charset filter，字集取的是 PingFangSC 所收的 3 万多字。对应配置可以从 actions 里取用。

**Reply by liangkaichun** [reply]
太牛了！这就下载使用，感谢！

**Reply by liangkaichun** [reply]
完美！

我已经放弃了之前自己修改的精简配置，直接使用最新的配置文件，选择辅筛方案，和我的需求完美匹配。

我自己一直在使用的一些特性：
	- 自然双拼方案
	- 直接辅助码筛选、造词
	- ；‘ 选择第二三候选项
	- ，。翻页
	- [  ] 以词定字（用了雾凇里的一个lua文件）

Mac上用鼠须管、安卓手机用小企鹅、iPhone上用仓输入法，同步的问题暂时不考虑。

**Reply by bosbyj** [reply]
翻页，我现在用的自带的emacs快捷键，ctrl p ctrl n 很好用。

以词定字不错啊，方括号。win11自带拼音输入法也有这个功能。

**Reply by liangkaichun** [reply]
这个方案有一些性能问题，CPU占用会比关掉过滤后高一倍，已经有比较明显的卡顿了

**Reply by davidlauhn** [reply]
跟我的习惯一样，看来都是老自然码用户，我还习惯用 jv qv xv yv，因为只支持魔然和魔然整句，所以自己改了 moran_aux.schema.yaml

把

`      - moran:/user_sentence_top?`

改成了 

`      - moran:/algebra/user_sentence_top?`

不知道会有什么后果，目前没有发现问题

**Comment by liangkaichun** [comment]
“新增：輔篩模式（moran_aux）。可在輸入末尾打一位或兩位輔助碼以篩選第一個詞語。請注意，該模式依賴於最新版本的 Lua 插件！**如果上屏後無法清空**。使用例：”

试用了这个辅筛模式，输入得辅助码在筛选后还存在，没有实现视频演示的效果，这个是什么原因？

“**如果上屏後無法清空**”，另外，这句话好像没说完。

**Reply by ksqsf** [reply]
我编辑了release这边的, discussion好像没同步过去...

辅助码没清掉就是librime-lua版本低的缘故, 依赖于2023年8月的一个新功能, 而目前鼠须管尚未发布可以使用该功能的版本.

**Reply by liangkaichun** [reply]
安装了鼠须管最新的action版本，部署的时候应用崩溃了，除了自己clone源码编译，还有什么办法可以更新librime-lua吗？

**Reply by ksqsf** [reply]
鼠须管的 action 的确有问题，我之后看看能不能修一下。

目前，如果不想自己编译还想用最新版 librime 的话，可以考虑使用 [fcitx5-macos](https://github.com/fcitx-contrib/fcitx5-macos-installer/blob/master/README.zh-CN.md)，但是和仓一样，f5m 目前没有打包 charcode。

**Reply by liangkaichun** [reply]
感谢!

找了多个鼠须管的fork，都没有提供可用的release版本，我先用小企鹅来体验一下辅筛模式吧，感谢！

**Reply by ksqsf** [reply]
https://github.com/rime/squirrel/releases/tag/latest 鼠须管的nightly build已上线，经测试可用。（然而还是没有 charcode……）

**Reply by liangkaichun** [reply]
太好了，我这就升级试试！

看了一下包的大小，和16.2版本差不多，为啥会少了charcode 呢？

我现在用小企鹅，除了不能自定义外观，其他的体验基本一致。

**Comment by kuqquk** [comment]
在Windows上使用小狼毫v0.15。下载了最新版本的魔然方案，直接复制粘贴到用户文件夹。但发现设置为“简体”时，选词列表里简体和繁体字会同时出现（设置为“繁体”时仅有繁体字）。
![Screenshot 2024-04-16](https://github.com/ksqsf/rime-moran/assets/51459108/54d73b97-88ba-441a-9dd4-e3d9fb1860f8)

但在简体模式下，似乎并不是所有的词都会有繁简同时出现的情况，比如在简体下输入“发财”则只有简体“发财”。
![Screenshot2](https://github.com/ksqsf/rime-moran/assets/51459108/f602ee19-f5ec-4428-ac81-58e7e9198100)

**Reply by ksqsf** [reply]
请问之前是否使用过繁体版？这可能是用户词库干扰，繁体版的用户词库也会记录繁体词条。若是，建议清理一下。

**Reply by kuqquk** [reply]
我之前极偶尔的用过繁体，清理了moran.userdb目录后，问题得到解决了。

**Comment by liangkaichun** [comment]
> 实现了 lua 版本的 charset filter，字集取的是 PingFangSC 所收的 3 万多字。对应配置可以从 actions 里取用。

这个方案有一些性能问题，CPU占用会比关掉过滤后高一倍，已经有比较明显的卡顿了

**Comment by ndosjinogso** [comment]
非常感谢，我自己一直在使用的：

自然码双拼方案
直接辅助码筛选
 
一个问题，打字时如果需要打出“V1，V5”，V会自动变成逗号“，” 不知道如何解决？

**Reply by ksqsf** [reply]
没太理解这个问题……我使用 Shift-v 1 , Shift-v 5 空格 可以得到「V1，V5」，是预期的吗？

**Reply by ndosjinogso** [reply]
是的，不知道为什么，我按shift V 然后按空格就会变逗号

**Comment by ndosjinogso** [comment]
求助，大家在WORD，Notes等文字输入软件中，有没有遇到打出的字母不上屏，而是在备选框的现象？要怎么解决呢？ 
现在如图所示：
<img width="202" alt="Screenshot 2024-05-01 at 10 56 32 PM" src="https://github.com/ksqsf/rime-moran/assets/140030014/d151bd5d-3c56-469b-a9dd-9fd8f38affe0">

对比正常情况：
<img width="103" alt="Screenshot 2024-05-01 at 10 59 45 PM" src="https://github.com/ksqsf/rime-moran/assets/140030014/5de788a5-2d42-4b5e-8c9d-bad6c51f1006">

**Reply by ksqsf** [reply]
这个行为是 `inline_preedit` 控制的，目前的默认设置已经是 true 了。可能鼠须管觉得某些程序没法 inline preedit，就放到候选框上了。

**Reply by ndosjinogso** [reply]
我发现在wechat dark里面的inline_preedit 默认为false，所以每次晚上发生这个问题。如果有同样问题的朋友可以把它改一下就好了。

**Comment by davidlauhn** [comment]
用的小狼毫 ，简体版，魔然+辅筛，但是好像有点小问题，比如 「她们」，输入 tamfn 第一候选是 「她们」，想空格上屏却出现「她们n」，然后再空格后就直接「她们你」上屏了

update: 貌似安装了 nightly build 之后可以了，可能还是上面提到的 librime 的问题

**Reply by bosbyj** [reply]
之前遇到同样的情况。

请问大家爱用的是哪个？

- 小狼毫稳定版 0.15.0
- 官方nightly
- fxliang fork release

**Reply by davidlauhn** [reply]
以前一直用0.15.0，昨天才试了 nightly build，习惯了

**Comment by davidlauhn** [comment]
辅筛模式在小企鹅输入法中单字无法使用辅助码，不知道是不是个案

比如 「源」 打 yrd，出来的是 「元旦」

另外，自定义的逗号句号翻页键，在通过 [Scrcpy ](https://github.com/Genymobile/scrcpy) 连电脑使用小企鹅输入法的时候无法用逗号句号翻页，不知道是小企鹅的问题还是啥，之前可以的，有些搞不懂

以上的问题刚装上小企鹅的时候都是可以的，但是过了几天发现不行了，今天电脑上的rime也不行了，胡乱捣鼓了一下，发现又行了，但小企鹅还是不行

**Reply by bosbyj** [reply]
#64 已经解决了，恭喜

**Reply by davidlauhn** [reply]
可能还是小企鹅的问题，小狼毫的辅筛模式现在是没有问题了，自定义了 moran.custom/moran_aux.custom/key_bindings.custom, 只是可能由于简体词库还不是很完美，有些单字打起来不是特别的顺手，但已经非常非常满意了，感谢 @ksqsf 

手机上的话，目前单字也能使用辅助码了，除了不能使用翻页快捷键，感觉应该是小企鹅对快捷键的支持有问题

**Reply by ksqsf** [reply]
@davidlauhn 任何词库和功能问题和建议都可以直接开Issue反馈，谢谢了！（PR也行）

**Reply by davidlauhn** [reply]
OK, 收到

**Comment by bosbyj** [comment]
距离第一个release 已经快1周年啦，非常优秀的项目。

请问大佬大约啥时候发2.0版？我就迁移过去长期固定使用了。

基于moran编排思路，替换成简约版的小鹤配置，已经积攒1个月的个人词库，愉快使用中。

**Reply by ksqsf** [reply]
暂时没有定期发布的计划，达到比较好的品质时再发布也不迟~ 不急这一会
